.github/workflows/main.yml

```diff
@@ -1,40 +1,48 @@
 name: main
-
 on:
-    push:
-        branches:
-            - main
-    pull_request:
+  pull_request:
+    types: [opened, reopened, synchronize]
+    branches: [main]
+  push:
+    branches: [main]
 
 concurrency:
-    group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
-    cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
 
 jobs:
-    build:
-        runs-on: ubuntu-latest
-
-        steps:
-            - name: Checkout
-              uses: actions/checkout@v4
-
-            - name: Install Rust
-              uses: dtolnay/rust-toolchain@stable
-              with:
-                  toolchain: stable
-                  components: rustfmt, clippy
-
-            - name: Cache
-              uses: Swatinem/rust-cache@v2
-
-            - name: Build
-              run: cargo build --workspace --all-targets
-
-            - name: Run tests
-              run: cargo test --workspace --all-targets
-
-            - name: Run clippy
-              run: cargo clippy --workspace --all-targets -- -D warnings
-
-            - name: Run fmt check
-              run: cargo fmt -- --check
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v2.7.3
+      - run: make build
+
+  format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: taiki-e/cache-cargo-install-action@v2.0.1
```

Comment:
What does this action do exactly? Why isn't it using `dtolnay/rust-toolchain@stable` like in the lint job? This allows to install extra crates maybe?

Reply:
https://github.com/taiki-e/cache-cargo-install-action?tab=readme-ov-file#cache-cargo-install-action

> This installs the specified crate using cargo install and caches the installed binaries using [actions/cache](https://github.com/actions/cache). If binaries for the specified crate are already cached, restore the cache instead of calling cargo install.

I found this action as I was looking for cargo related actions. I hadn't seen you had selected `dtolnay/rust-toolchain` and `Swatinem/rust-cache@v2.7.3` already.

I think that 
```yaml
format:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: taiki-e/cache-cargo-install-action@v2.0.1
        with:
          tool: dprint
      - run: make check
```
achieves in 3 steps what 
```yaml
format:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: dtolnay/rust-toolchain@stable
      - uses: Swatinem/rust-cache@v2.7.3
      - run: cargo install dprint
      - run: make check
```
does in 5.

We could stick to  dtolnay/rust-toolchain and Swatinem/rust-cache if you prefer

Reply:
Ahh looks great to me! Should we use `taiki-e/install-action` instead tho?

Reply:
based on the README maybe.
But it is not clear to me if taiki-e/install-action is really caching stuff like taiki-e/cache-cargo-install-action does